packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -960,7 +960,10 @@ export class Stack extends Construct implements ITaggable {
       throw new Error(`Context value '${cxapi.TARGET_PARTITIONS}' should be a list of strings, got: ${JSON.stringify(partitions)}`);
     }
 
-    const lookupMap = partitions ? RegionInfo.limitedRegionMap(factName, partitions) : RegionInfo.regionMap(factName);
+    const lookupMap =
+      partitions !== undefined && partitions !== 'undefined'
+        ? RegionInfo.limitedRegionMap(factName, partitions)
+        : RegionInfo.regionMap(factName);
```

Comment:
These changes belong to a commit that was submitted in a separate PR: https://github.com/aws/aws-cdk/pull/23023

I have to rebase-on-top/include this commit in this PR because without it, the added integration test will fail.